Framework/Core/include/Framework/DataProcessingDevice.h

```diff
@@ -42,7 +46,7 @@ class DataProcessingDevice : public FairMQDevice {
   AlgorithmSpec::ProcessCallback mStatelessProcess;
   AlgorithmSpec::ErrorCallback mError;
   std::unique_ptr<ConfigParamRegistry> mConfigRegistry;
-  ServiceRegistry mServiceRegistry;
+  ServiceRegistry& mServiceRegistry;
```

Comment:
@ktf, probably it was not intended, to make a copy here, so I changed to storing a reference